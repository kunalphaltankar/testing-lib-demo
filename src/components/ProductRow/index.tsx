import { IProduct } from "../../models";

function ProductRow({
  product: { name, price, stocked },
}: {
  product: IProduct;
}) {
  return (
    <tr>
      <td>{stocked ? name : <span style={{ color: 'red' }}>{name}</span>}</td>
      <td>{price}</td>
    </tr>
  );
}

export default ProductRow;