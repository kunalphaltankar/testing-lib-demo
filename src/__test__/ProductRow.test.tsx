import { render, screen } from '@testing-library/react';
import ProductRow from '../components/ProductRow';
import PRODUCTS from '../models/stub/PRODUCTS';

const product = PRODUCTS[1];
const notInStockProduct = PRODUCTS[2];

const tbody = document.createElement('tbody');

describe('Product Row', () => {
  it('should render stocked product', () => {
    const { container } = render(<ProductRow product={product} />, {
      container: document.body.appendChild(tbody),
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with unstocked product', () => {
    const { container } = render(<ProductRow product={notInStockProduct} />, {
      container: document.body.appendChild(tbody),
    });

    expect(container).toMatchSnapshot();

    const span = screen.getByText(/Basketball/i);
    expect(span).toHaveStyle('color: red;');
  });
});
