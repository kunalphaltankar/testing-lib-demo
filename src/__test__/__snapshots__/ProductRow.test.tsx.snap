// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Row should render stocked product 1`] = `
<tbody>
  <tr>
    <td>
      Baseball
    </td>
    <td>
      $9.99
    </td>
  </tr>
</tbody>
`;

exports[`Product Row should render with unstocked product 1`] = `
<tbody>
  <tr>
    <td>
      <span
        style="color: red;"
      >
        Basketball
      </span>
    </td>
    <td>
      $29.99
    </td>
  </tr>
</tbody>
`;
